test/jdk/java/nio/channels/DatagramChannel/SelectWhenRefused.java

```diff
@@ -94,9 +131,38 @@ public static void main(String[] args) throws IOException {
     }
 
     static void sendDatagram(DatagramChannel dc, SocketAddress remote)
-        throws IOException
-    {
-        ByteBuffer bb = ByteBuffer.wrap("Greetings!".getBytes());
+            throws IOException {
+        ByteBuffer bb = ByteBuffer.wrap(GREETINGS_MESSAGE.getBytes());
         dc.send(bb, remote);
     }
+
+    static boolean checkUnexpectedWakeup(Set<SelectionKey> selectedKeys) {
+        System.out.format("Received %d keys%n", selectedKeys.size());
+        for (SelectionKey key : selectedKeys) {
+            if (!key.isValid() || !key.isReadable()) {
+                System.out.println("Invalid or unreadable key: " + key);
+                continue;
+            }
+            try {
+                System.out.println("Attempting to read datagram from key: " + key);
+                DatagramChannel datagramChannel = (DatagramChannel) key.channel();
+                ByteBuffer buf = ByteBuffer.allocate(100);
+                SocketAddress sa = datagramChannel.receive(buf);
+                buf.flip();
+                byte[] bytes = new byte[buf.remaining()];
+                buf.get(bytes);
+                String message = new String(bytes);
+                if (sa != null) {
```

Comment:
L151-154 should be moved into the if (sa != null) { ... } as there are no bytes read when the sender is null.